code/game/objects/structures/stool_bed_chair_nest/guillotine.dm

```diff
@@ -0,0 +1,256 @@
+				"You attempt to dislodge \the [src]'s stocks.")
+			if(do_after(M, src, 300))
+				M.visible_message(\
+					"<span class='warning'>\The [M] dislodges \the [src]'s stocks and climbs out of \the [src]!</span>",\
+					"You dislodge \the [src]'s stocks and climb out of \the [src].")
+			else
+				return
+		else
+			M.visible_message(\
+				"<span class='notice'>\The [M] climbs out of \the [src].</span>",\
+				"You climb out of \the [src].")
+	unlock_atom(M)
+
+	add_fingerprint(user)
+
+/obj/structure/bed/guillotine/buckle_mob(mob/M, mob/user)
+	if(!Adjacent(user) || user.incapacitated() || istype(user, /mob/living/silicon/pai) || user == victim)
+		return
+
+	if(!ismob(M) || !M.Adjacent(user)  || M.locked_to)
+		return
+
+	if(!anchored)
+		to_chat(user, "<span class='warning'>\The [src] needs to be anchored first.</span>")
+		return
+	if(bladedown)
+		to_chat(user, "<span class='warning'>You can't fit \the [M] into \the [src] while the blade is down.</span>")
+		return
+	if(!open)
+		to_chat(user, "<span class='warning'>You can't place \the [M] into \the [src] while its stocks are closed.</span>")
+		return
+
+	for(var/mob/living/L in get_locked(lock_type))
+		if(L.stat)
+			to_chat(user, "<span class='warning'>There is still a body inside \the [src].</span>")
+		else
+			to_chat(user, "<span class='warning'>There is already someone inside \the [src].</span>")
+		return
+
+	if(user.size <= SIZE_TINY) //Fuck off mice
+		to_chat(user, "<span class='warning'>You are too small to do that.</span>")
+		return
+
+	if(!ishuman(M))
+		return
+
+	if(M == user)
+		M.visible_message(\
+			"<span class='notice'>\The [M] climbs into \the [src]!</span>",\
+			"You climb into \the [src].")
+	else
+		M.visible_message(\
+			"<span class='warning'>\The [M] is placed in \the [src] by \the [user]!</span>",\
+			"<span class='danger'>You are placed in \the [src] by \the [user].</span>")
+	add_fingerprint(user)
+
+	lock_atom(M, lock_type)
+
+/obj/structure/bed/guillotine/lock_atom(var/atom/movable/AM, var/datum/locking_category/category = /datum/locking_category)
+	. = ..()
+	if(.)
+		victim = AM
+		AM.dir = NORTH
+		var/matrix/M = matrix()
+		M.Turn(180)
+		M.Scale(1,0.5)
+		AM.transform = M
+		update_icon()
+
+/obj/structure/bed/guillotine/unlock_atom(var/atom/movable/AM)
+	. = ..()
+	if(.)
+		AM.dir = SOUTH
+		var/matrix/M = AM.transform
+		M.Turn(180)
+		if(!victim.lying)
+			M.Scale(1,2)
+		AM.transform = M
+		victim = null
+		update_icon()
+
+/obj/structure/bed/guillotine/attackby(obj/item/weapon/W, mob/user)
+	if(user == victim)
+		return
+	if(iswrench(W))
+		if(anchored)
+			if(victim)
+				to_chat(user, "<span class='warning'>You can't unsecure \the [src] from the floor while someone's inside it!</span>")
+				return
+			to_chat(user, "You unsecure \the [src] from the floor.")
+		else
+			to_chat(user, "You secure \the [src] to the floor.")
+		playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+		anchored = !anchored
```

Comment:
The proc for a generic wrench based unanchor is obj level but I can't recall the name right now.